### PR TITLE
Different Photon Energy Ranges

### PR DESCRIPTION
**Pull request by slacAWallace**
Needed a way to have different photon energy ranges for each beamline.

### CODE REVIEW

**Comment by slacAWallace** [comment]
@klauer @ZLLentz Compile defines are cool. Compile defines in 4024 will be even more powerful.

https://infosys.beckhoff.com/content/1033/tc3_plc_intro/36028799548759947.html?id=7862851008802573676

File: lcls-twincat-pmps/PMPS/HelperFunctions/F_eVRangeCalculator.TcPOU [hunk 1]
```diff
@@ -27,38 +27,47 @@ VAR
...
+	Boundaries : ARRAY [0..PMPS_GVL.g_cBoundaries] OF REAL;
 END_VAR
 ]]></Declaration>
     <Implementation>
-      <ST><![CDATA[
+      <ST><![CDATA[{IF defined (L)}
+	Boundaries := PMPS_GVL.g_areVBoundariesL;
+{ELSIF defined (K)}
+	Boundaries := PMPS_GVL.g_areVBoundariesK;
+{END_IF}
```
Comment (ZLLentz) [inline lcls-twincat-pmps/PMPS/HelperFunctions/F_eVRangeCalculator.TcPOU:40]: Interesting! Where would we go about defining L or K in user code?

**Comment by slacAWallace** [comment]
image

**Comment by ZLLentz** [comment]
Also, you should consider carefully the ramifications of locking yourself into a single-character compilation define. Not that I have a better suggestion.

**Comment by klauer** [comment]
Innnteresting...

**Pull request merged**